build_all/base-libraries/AzureIoTHub/README.md

```diff
@@ -1,7 +1,135 @@
+This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
+
 #### Azure IoT Hub Library source files
 
+
+# AzureIoTHub - Azure IoT Hub library for Arduino
+
 This is the location of the Arduino-specific source files for the
+
 [AzureIoTHub Arduino published library](https://github.com/Azure/azure-iot-arduino). 
 
+This library is a port of the [Microsoft Azure IoT device SDK for C](https://github.com/Azure/azure-iot-sdks/blob/master/c/readme.md) to Arduino. It allows you to use several Arduino compatible boards with Azure IoT Hub. Please submit any contribution directly to [azure-iot-sdks](https://github.com/Azure/azure-iot-sdks).
+
+Currently supported hardware:
+
+- ESP8266 based boards with [esp8266/arduino](https://github.com/esp8266/arduino)
+
+  - SparkFun [Thing](https://www.sparkfun.com/products/13711)
+
+  - Adafruit [Feather Huzzah](https://www.adafruit.com/products/2821)
+  
+- ESP32 based boards with [espressif/arduino-esp32](https://github.com/espressif/arduino-esp32)
+  
+  - Adafruit [HUZZAH32](https://www.adafruit.com/product/3405)
+
+## Prerequisites
+
+You should have the following ready before beginning with any board:
+
+-   [Setup your IoT hub](https://github.com/Azure/azure-iot-device-ecosystem/blob/master/setup_iothub.md)
+
+-   [Provision your device and get its credentials](https://github.com/Azure/azure-iot-device-ecosystem/blob/master/setup_iothub.md#create-new-device-in-the-iot-hub-device-identity-registry)
+
+-   [Arduino IDE](https://www.arduino.cc/en/Main/Software)
+
+-   Generate the Libraries by executing the make_sdk.cmd file with the folder to build to as arg 1, and the board you are using (esp8266, esp32) as arg 2 *OR*:
```

Comment:
there should probably be a link to the `make_sdk.cmd` since if I were looking in this directory, I don't see that file. It would be confusing where to go from here.

Reply:
nit: lets put esp8266 and esp32 in `` ` ``. So it's formatted as `esp8266` and `esp32`.

Reply:
Agreed, I'm reformatting it a bit, I left the location of the make_sdk.cmd out of this how-to.

Reply:
Updated for clarity.